tests/sentry/snuba/metrics/test_query_builder.py

```diff
@@ -1036,3 +1036,114 @@ def test_valid_latest_release_alias_filter(self):
                 rhs=["bar"],
             )
         ]
+
+
+class ResolveTagsTestCase(TestCase):
+    def setUp(self):
+        self.org_id = ORG_ID
+        self.use_case_id = UseCaseKey.PERFORMANCE
+
+    def test_resolve_tags_with_unary_tuple(self):
+        transactions = ["/foo", "/bar"]
+
+        indexer.record(self.use_case_id, self.org_id, string="transaction")
+
+        for transaction in transactions:
+            indexer.record(use_case_id=self.use_case_id, org_id=self.org_id, string=transaction)
+
+        resolved_query = resolve_tags(
+            self.use_case_id,
+            self.org_id,
+            Condition(
+                lhs=Function(
+                    function="tuple",
+                    parameters=[
+                        Column(
+                            name="tags[transaction]",
+                        )
+                    ],
+                ),
+                op=Op.IN,
+                rhs=Function(
+                    function="tuple",
+                    parameters=[(transaction,) for transaction in transactions],
+                ),
+            ),
+        )
+
+        assert resolved_query == Condition(
+            lhs=Function(
+                function="tuple",
+                parameters=[
+                    Column(
+                        name=resolve_tag_key(self.use_case_id, self.org_id, "transaction"),
+                    )
+                ],
+            ),
+            op=Op.IN,
+            rhs=Function(
+                function="tuple",
+                parameters=[
+                    (resolve_tag_value(self.use_case_id, self.org_id, transaction),)
+                    for transaction in transactions
+                ],
+            ),
+        )
+
+    def test_resolve_tags_with_binary_tuple(self):
+        tags = [("/foo", "ios"), ("/bar", "android")]
+
+        indexer.record(self.use_case_id, self.org_id, string="transaction")
+        indexer.record(self.use_case_id, self.org_id, string="platform")
+
+        for transaction, platform in tags:
+            indexer.record(use_case_id=self.use_case_id, org_id=self.org_id, string=transaction)
+            indexer.record(use_case_id=self.use_case_id, org_id=self.org_id, string=platform)
```

Comment:
Can you combine those maybe like this? 
```suggestion
        for transaction, platform in [("transaction", "platform")] + tags:
            indexer.record(use_case_id=self.use_case_id, org_id=self.org_id, string=transaction)
            indexer.record(use_case_id=self.use_case_id, org_id=self.org_id, string=platform)
```

Reply:
How does it achieve better readability?